beam/beam/scan/user_login.py

```diff
@@ -0,0 +1,34 @@
+import frappe
+from beam.beam.scan import get_barcode_context
+
+
+@frappe.whitelist(allow_guest=True)
+def scan_login(barcode):
+	client_ip = (
+		frappe.local.request.headers.get("X-Forwarded-For") or frappe.local.request.remote_addr
+	)
+	print(f"Barcode: {barcode}, IP: {client_ip}")
+
+	user = get_barcode_context(barcode)
+	if not user:
+		frappe.throw("Wrong barcode", title="Login Error")
+
+	if user["doc"].doctype != "User":
+		frappe.throw("Wrong barcode", title="Login Error")
+
+	user_doc = frappe.get_doc("User", user["doc"].name)
+	if not user_doc:
+		frappe.throw("User doesn't exist", title="Login Error")
```

Comment:
`frappe.get_doc` will fail with an error anyway if the record doesn't exist, so this check can be removed